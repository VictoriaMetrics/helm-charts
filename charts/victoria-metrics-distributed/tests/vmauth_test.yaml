suite: vmauth templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - vmauth-*.yaml
  - per-az/vmauth-*.yaml
tests:
  - it: vmauth should match snapshot
    set:
      write:
        global:
          vmauth:
            spec:
              image:
                tag: v1.0.2
      read:
        global:
          vmauth:
            spec:
              image:
                tag: v1.0.3
      zoneTpl:
        read:
          perZone:
            vmauth:
              spec:
                image:
                  tag: v1.0.4
          crossZone:
            vmauth:
              spec:
                image:
                  tag: v1.0.5
        write:
          vmauth:
            spec:
              image:
                tag: v1.0.6
        cluster:
          spec:
            image:
              tag: v1.0.7
        vmagent:
          spec:
            image:
              tag: v1.0.8
        
    asserts:
      - matchSnapshot: {}
