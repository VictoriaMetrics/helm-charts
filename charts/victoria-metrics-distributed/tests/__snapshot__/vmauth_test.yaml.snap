vmauth should match snapshot:
  1: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-read-balancer-zone-eu-1
      namespace: NAMESPACE
    spec:
      affinity: {}
      extraArgs:
        discoverBackendIPs: "true"
      nodeSelector:
        topology.kubernetes.io/zone: zone-eu-1
      port: "8427"
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
      unauthorizedAccessConfig:
        - src_paths:
            - /select/.+
          url_prefix:
            - http://vmselect-vmcluster-zone-eu-1.NAMESPACE.svc.cluster.local.:8481
  2: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-read-balancer-zone-us-1
      namespace: NAMESPACE
    spec:
      affinity: {}
      extraArgs:
        discoverBackendIPs: "true"
      nodeSelector:
        topology.kubernetes.io/zone: zone-us-1
      port: "8427"
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
      unauthorizedAccessConfig:
        - src_paths:
            - /select/.+
          url_prefix:
            - http://vmselect-vmcluster-zone-us-1.NAMESPACE.svc.cluster.local.:8481
  3: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-read-proxy-zone-eu-1
      namespace: NAMESPACE
    spec:
      affinity: {}
      nodeSelector:
        topology.kubernetes.io/zone: zone-eu-1
      port: "8427"
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
      unauthorizedAccessConfig:
        - load_balancing_policy: first_available
          retry_status_codes:
            - 503
          src_paths:
            - /select/.+
          url_prefix:
            - http://vmauth-vmauth-read-balancer-zone-eu-1.NAMESPACE.svc.cluster.local.:8427
            - http://vmauth-vmauth-read-balancer-zone-us-1.NAMESPACE.svc.cluster.local.:8427
  4: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-read-proxy-zone-us-1
      namespace: NAMESPACE
    spec:
      affinity: {}
      nodeSelector:
        topology.kubernetes.io/zone: zone-us-1
      port: "8427"
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
      unauthorizedAccessConfig:
        - load_balancing_policy: first_available
          retry_status_codes:
            - 503
          src_paths:
            - /select/.+
          url_prefix:
            - http://vmauth-vmauth-read-balancer-zone-us-1.NAMESPACE.svc.cluster.local.:8427
            - http://vmauth-vmauth-read-balancer-zone-eu-1.NAMESPACE.svc.cluster.local.:8427
  5: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-write-balancer-zone-eu-1
      namespace: NAMESPACE
    spec:
      affinity: {}
      extraArgs:
        discoverBackendIPs: "true"
      nodeSelector:
        topology.kubernetes.io/zone: zone-eu-1
      port: "8427"
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
      unauthorizedAccessConfig:
        - src_paths:
            - /insert/.+
          url_prefix:
            - http://vminsert-vmcluster-zone-eu-1.NAMESPACE.svc.cluster.local.:8480
  6: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-write-balancer-zone-us-1
      namespace: NAMESPACE
    spec:
      affinity: {}
      extraArgs:
        discoverBackendIPs: "true"
      nodeSelector:
        topology.kubernetes.io/zone: zone-us-1
      port: "8427"
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: ScheduleAnyway
      unauthorizedAccessConfig:
        - src_paths:
            - /insert/.+
          url_prefix:
            - http://vminsert-vmcluster-zone-us-1.NAMESPACE.svc.cluster.local.:8480
  7: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-global-read-RELEASE-NAME-vm-distributed
      namespace: NAMESPACE
    spec:
      port: "8427"
      unauthorizedAccessConfig:
        - load_balancing_policy: first_available
          src_paths:
            - /select/.+
          url_prefix:
            - http://vmauth-vmauth-read-proxy-zone-eu-1.NAMESPACE.svc.cluster.local.:8427
            - http://vmauth-vmauth-read-proxy-zone-us-1.NAMESPACE.svc.cluster.local.:8427
  8: |
    apiVersion: operator.victoriametrics.com/v1beta1
    kind: VMAuth
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vm-distributed
        helm.sh/chart: victoria-metrics-distributed-0.1.1
      name: vmauth-global-write-RELEASE-NAME-vm-distributed
      namespace: NAMESPACE
    spec:
      port: "8427"
      unauthorizedAccessConfig:
        - src_paths:
            - /api/v1/write
            - /prometheus/api/v1/write
            - /write
            - /api/v1/import
            - /api/v1/import/.+
          url_prefix:
            - http://vmagent-vmagent-zone-eu-1.NAMESPACE.svc.cluster.local.:8429
            - http://vmagent-vmagent-zone-us-1.NAMESPACE.svc.cluster.local.:8429
